File: src/main/java/guru/xgm/language/lni/formatting/block/LniItemBlock.java
package guru.xgm.language.lni.formatting.block;

import com.intellij.formatting.Alignment;
import com.intellij.formatting.Block;
import com.intellij.formatting.Indent;
import com.intellij.lang.ASTNode;
import com.intellij.psi.codeStyle.CodeStyleSettings;
import guru.xgm.language.lni.formatting.LniCodeStyleSettings;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import static com.intellij.psi.formatter.FormatterUtil.isOneOf;
import static guru.xgm.language.lni.psi.LniTypes.PROPERTY;

public class LniItemBlock extends LniBlock {
    public LniItemBlock(ASTNode myNode, Alignment myAlignment, Indent myIndent, CodeStyleSettings codeStyleSettings) {
        super(myNode, myAlignment, myIndent, codeStyleSettings);
        final var lni = codeStyleSettings.getCustomSettings(LniCodeStyleSettings.class);
        propertyEqAlignment = lni.AT_PROPERTY_EQ ? Alignment.createAlignment(true, Alignment.Anchor.LEFT) : null;
    }

    @Nullable Alignment propertyEqAlignment;

    @Override
    public Block makeSubBlock(@NotNull ASTNode childNode) {
        if (isOneOf(childNode, PROPERTY))
            return new LniPropertyBlock(this, childNode, myAlignment, myIndent, myCodeStyleSettings);

        return new LniBlock(childNode, myAlignment, myIndent, myCodeStyleSettings);
    }
}

File: src/main/kotlin/raft/war/language/lni/formatting/block/LniItemBlock.kt
package raft.war.language.lni.formatting.block

import com.intellij.formatting.Alignment
import com.intellij.formatting.Block
import com.intellij.formatting.Indent
import com.intellij.lang.ASTNode
import com.intellij.psi.codeStyle.CodeStyleSettings
import com.intellij.psi.formatter.FormatterUtil
import raft.war.language.lni.formatting.LniCodeStyleSettings
import raft.war.language.lni.psi.LniTypes

class LniItemBlock(myNode: ASTNode, myAlignment: Alignment?, myIndent: Indent?, codeStyleSettings: CodeStyleSettings) :
    LniBlock(
        myNode, myAlignment, myIndent, codeStyleSettings
    ) {
    var propertyEqAlignment: Alignment?

    init {
        val lni = codeStyleSettings.getCustomSettings(LniCodeStyleSettings::class.java)
        propertyEqAlignment = if (lni.AT_PROPERTY_EQ) Alignment.createAlignment(true, Alignment.Anchor.LEFT) else null
    }

    override fun makeSubBlock(childNode: ASTNode): Block {
        if (FormatterUtil.isOneOf(childNode, raft.war.language.lni.psi.LniTypes.PROPERTY)) return LniPropertyBlock(
            this,
            childNode,
            myAlignment,
            myIndent,
            myCodeStyleSettings
        )

        return LniBlock(childNode, myAlignment, myIndent, myCodeStyleSettings)
    }
}
